### PR TITLE
Make challengeName optional in AWS Cognito User Pool DefineAuthChallengeTriggerEvent response.

### PR DESCRIPTION
**Pull request by mawallace**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-define-auth-challenge.html shows examples of setting the response without a `challengeName`. The `challengeName` is only set when issuing a challenge, not when indicating successful/failed auth.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/aws-lambda/trigger/cognito-user-pool-trigger/define-auth-challenge.d.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export interface DefineAuthChallengeTriggerEvent extends BaseTriggerEvent<"Defin
         clientMetadata?: StringMap | undefined;
     };
     response: {
-        challengeName: string;
+        challengeName: string | undefined;
```
Comment (darbio) [inline types/aws-lambda/trigger/cognito-user-pool-trigger/define-auth-challenge.d.ts:15]: Can you change this to `challengeName?: string | undefined` as per the rest of the file.

**Review by darbio — CHANGES_REQUESTED** [review]
I read the documentation and in the example it does show a response where the challenge name property is not set on the response, which does look like the  proposed changes are valid.

This is a breaking change - the change in type from `string` to `string | undefined` may cause some people some typing dramas, but this is one of those types which isn't usually created by anyone and instead passed in by AWS, and it won't break the setting of the parameter, which would form the vast majority of the uses (it would be unlikely that anything else is happening here in production code? I'm sure we'll break some tests 💩)

Please edit the tests so that they reflect the change in type, and run successfully, and I will re-review the code to get this merged.

**Comment by mawallace** [comment]
@darbio -- thanks for reviewing this. I have made the change you requested and updated the test.

I agree with your comment "this is one of those types which isn't usually created by anyone and instead passed in by AWS, and it won't break the setting of the parameter". Though, just for context, my motivation for making this change is to actually create the type, since I think it can make code clearer. That is, it would allow me to create constants like:

```
const ISSUE_CUSTOM_CHALLENGE_RESPONSE = {
  issueTokens: false,
  failAuthentication: false,
  challengeName: 'CUSTOM_CHALLENGE',
};

const PASS_AUTHENTICATION_RESPONSE = {
  issueTokens: true,
  failAuthentication: false,
};

const FAIL_AUTHENTICATION_RESPONSE = {
  issueTokens: false,
  failAuthentication: true,
};
```

And then do something like

```
if (...) {
  event.response = ISSUE_CUSTOM_CHALLENGE_RESPONSE;
}
```

in my lambda trigger.

I know I could already have functions that mutate the event/response in a similar way, but I found setting the response like this easier to understand.

Thanks again.

**Review by sheetalkamat — APPROVED** [review]

**Comment by mawallace** [comment]
Ready to merge

**Pull request merged**